### PR TITLE
dotnet-watch: Enable nullable annotations on project level

### PR DESCRIPTION
**Pull request by tmat**
Enable nullable annotations on project level

### CODE REVIEW

**Comment by tmat** [comment]
@tlmii PTAL

File: src/BuiltInTools/DotNetWatchTasks/Utilities/CompilerFeatureRequiredAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        public const string RefStructs = nameof(RefStructs);
+
+        /// <summary>
+        /// The <see cref="FeatureName"/> used for the required members C# feature.
+        /// </summary>
+        public const string RequiredMembers = nameof(RequiredMembers);
+    }
+}
+
+#endif
```
Comment (sharwell) [inline src/BuiltInTools/DotNetWatchTasks/Utilities/CompilerFeatureRequiredAttribute.cs:45]: 📝 This should have an `#else` that type forwards the attribute for binary compat across builds

File: src/BuiltInTools/dotnet-watch/dotnet-watch.csproj [hunk 2]
```diff
@@ -32,39 +33,17 @@
...
   <ItemGroup>
-    <ProjectReference Include="..\BrowserRefresh\Microsoft.AspNetCore.Watch.BrowserRefresh.csproj"
-                      PrivateAssets="All"
-                      ReferenceOutputAssembly="false"
-                      SkipGetTargetFrameworkProperties="true"
-                      UndefineProperties="TargetFramework;TargetFrameworks"
-                      OutputItemType="Content"
-                      TargetPath="middleware\Microsoft.AspNetCore.Watch.BrowserRefresh.dll"
-                      CopyToOutputDirectory="PreserveNewest"/>
+    <ProjectReference Include="..\BrowserRefresh\Microsoft.AspNetCore.Watch.BrowserRefresh.csproj" PrivateAssets="All" ReferenceOutputAssembly="false" SkipGetTargetFrameworkProperties="true" UndefineProperties="TargetFramework;TargetFrameworks" OutputItemType="Content" TargetPath="middleware\Microsoft.AspNetCore.Watch.BrowserRefresh.dll" CopyToOutputDirectory="PreserveNewest" />
```
Comment (sharwell) [inline src/BuiltInTools/dotnet-watch/dotnet-watch.csproj:36]: 💡 Can you manually revert all the formatting changes?

**Review by tlmii — APPROVED** [review]
LGTM

**Pull request merged**